### PR TITLE
Doc: Add documentation for parametrize_all_cases and get_all_cases

### PR DESCRIPTION
**Pull request by eddiebergman**
Add documentation for the parameter types for `parametrize_all_cases` and `get_all_cases` with repsect to issue #257

### CODE REVIEW

**Comment by eddiebergman** [comment]
Hi @smarie, here's the clarification on the docs :)

File: docs/api_reference.md [hunk 1]
```diff
@@ -257,8 +257,9 @@ CaseFilter(filter_function: Callable)
 ### `@parametrize_with_cases`
 
 ```python
+CaseType = Union[Callable, Type, ModuleRef]
```
Comment (smarie) [inline docs/api_reference.md:260]:
```suggestion
CaseType = Union[Callable, Type, ModuleRef]

```

File: docs/api_reference.md [hunk 2]
```diff
@@ -335,8 +336,9 @@ Note that you can get the same contents directly by using the [`current_cases`](
 ### `get_all_cases`
 
 ```python
+CaseType = Union[Callable, Type, ModuleRef]
```
Comment (smarie) [inline docs/api_reference.md:339]:
```suggestion
CaseType = Union[Callable, Type, ModuleRef]

```

**Comment by smarie** [comment]
thanks @eddiebergman  !

**Pull request merged**